app/src/main/kotlin/pl/gov/mc/protego/information/PhoneInformation.kt

```diff
@@ -14,4 +21,16 @@ class PhoneInformation {
 
     val platform: String
         get() = "android"
+
+    val hasActiveInternetConnection: Boolean
+        get() {
+            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+            val mobileDataEnabled =  connectivityManager.allNetworks.any {
+                connectivityManager.getNetworkCapabilities(it)?.hasTransport(TRANSPORT_CELLULAR) == true
+            }
+
+            val wifiManager = context.getSystemService(Context.WIFI_SERVICE) as WifiManager?
+
+            return wifiManager?.isWifiEnabled ?: false || mobileDataEnabled
```

Comment:
Pytanie czy wystarczy nam takie sprawdzenie? Bo włączone ustawienia w settingsach niekoniecznie oznaczają, że użytkownik ma połączenie z internetem. Nie chcemy pingować backendu?

Reply:
To nam daje bardzo szybko informację czy mamy szansę mieć internet czy nie. Do zbierania danych internet nie jest potrzebny, natomiast każda akcja wykonywana przez uzytkownika takiego dostępu wymaga więc dosyć szybko dostaniemy błąd.

Reply:
https://github.com/ProteGO-app/android/issues/56